src/Controller/Component/AuthComponent.php

```diff
@@ -947,26 +947,46 @@ public function flash($message): void
         $this->Flash->set($message, $this->_config['flash']);
     }
 
+    /**
+     * Checks if there is an BaseAuthenticate instance.
+     *
+     * @return bool
+     */
+    public function hasAuthenticationProvider(): bool
```

Comment:
I think this is going to be a smoother change than throwing exceptions from `authenticationProvider()`.

Reply:
We need a non nullable return somewhere.
So you want to rather add OrFail() methods in these cases?

Reply:
This method seems redundant to me. One could just use `if ($this->Auth->user()) {` instead of `if ($this->Auth->hasAuthenticationProvider()) {`.